docs/design-document.md

```diff
@@ -39,24 +39,56 @@ default, Git defines two of refs: branches (heads) and tags. Git allows for the
 creation of other arbitrary refs that users can store other information as long
 as they are formatted using Git's object types.
 
+```
+Repository
+|
+|-- refs
+|   |
+|   |-- heads
+|   |   |-- master (refers to commit C)
+|   |   |-- feature-x (refers to commit E)
+|   |
+|   |-- tags
+|   |   |-- v1.0 (refers to commit B)
```

Comment:
Can you rename `commits` to be `objects` below and include a `v1.0` tag object that in turn refers to commit `B`?

Reply:
this message here would then say `refers to tag v1.0`